src/main/java/org/alfresco/rest/api/impl/QuickShareLinksImpl.java

```diff
@@ -618,6 +646,26 @@ private QuickShareLink getQuickShareInfo(NodeRef nodeRef, Map<String, Object> ma
         }
     }
 
+    private Map<String, Object> filterProps(Map<String, Object> properties, List<QName> toRemove)
+    {
+        Map<String, Object> filteredProps = new HashMap<>(properties);
+        List<String> propsToRemove = toRemove.stream().map(e -> e.toPrefixString(namespaceService)).collect(Collectors.toList());
+        filteredProps.keySet().removeAll(propsToRemove);
+        return filteredProps;
+    }
+
+    private Node getNode(Node node, NodeRef nodeRef, List<String> includeParam, Map<String, UserInfo> mapUserInfo)
+    {
+        if (node != null)
+        {
+            return node;
```

Comment:
This is an unusual caching mechanism. This should be done in the calling code instead of passing a node down to getNode()